theano/configdefaults.py

```diff
@@ -1524,11 +1536,24 @@ def _timeout_default():
 
 try:
     p_out = output_subprocess_Popen([config.cxx, '-dumpversion'])
+    cxx_version_str = p_out[0].strip().decode()
+except OSError:
+    # Typically means gcc cannot be found.
+    cxx_version_str = 'CXX_NOT_FOUND'
+
+try:
+    p_out = output_subprocess_Popen(['g++', '-dumpversion'])
```

Comment:
if config.cxx is a full path to a different g++ version then the default one, this won't grap the right version.